### PR TITLE
Add GEO import process

### PR DESCRIPTION
**Pull request by AnzeLovse**
## Checklist

This adds a GEO import process which imports all experiments belonging to a certain RNA-seq GEO series. Runs from a single experiment are joined together into a single sample. 

* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output fields have their ``re-save`` calls.

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ Unreleased
 Added
 -----
 - Add GEOparse to the ``resolwebio/common:2.7.0`` Docker image
+-  Add `geo-import` process
```
Comment (jkokosar) [inline docs/CHANGELOG.rst:16]: Fix double quotes

File: resolwe_bio/processes/workflows/geo_import.py [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+
+def construct_descriptor(metadata, sample_name):
+    """Construct a descriptor from sample metadata.
+
+    Dictionary with GEO metadata that matches the sample descriptor
+    schema is created. Atributes under general that have no
+    predetermined choices are matched with our naming if they
+    exist in the metadata. Other fileds with choices and the
+    experimental section are filled separely.
```
Comment (jkokosar) [inline resolwe_bio/processes/workflows/geo_import.py:52]: separely typo

File: resolwe_bio/processes/workflows/geo_import.py [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+        "resources": {
+            "cores": 1,
+            "memory": 8192,
+            "network": True,
+        },
+    }
+    data_name = "{{ gse_accession }}"
+    version = "1.0.0"
+    process_type = "data:geo"
+    category = "Pipeline"
```
Comment (jkokosar) [inline resolwe_bio/processes/workflows/geo_import.py:173]: Related SRA import processes are in `category = "Import"`. Add also this one to the same category, or create a new `SRA data` category

File: resolwe_bio/processes/workflows/geo_import.py [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+        if not re.match(r"(GSE\d{1,8})", inputs.gse_accession):
+            self.error(
+                f"GEO series accessions (GSE) are supported but {inputs.gse_accession} was provided."
+            )
+
+        try:
+            gse = GEOparse.get_GEO(geo=inputs.gse_accession, destdir="./")
+        except IOError as e:
+            print(e)
```
Comment (jkokosar) [inline resolwe_bio/processes/workflows/geo_import.py:294]: Is this print required or it was used just for the debugging?

File: resolwe_bio/processes/workflows/geo_import.py (hunk 3 above)
Comment (AnzeLovse) [reply resolwe_bio/processes/workflows/geo_import.py:173]: Good suggestion, I will add it to the `Import` category to keep it consistent with SRA import.

File: resolwe_bio/processes/workflows/geo_import.py (hunk 4 above)
Comment (AnzeLovse) [reply resolwe_bio/processes/workflows/geo_import.py:294]: This is redundant as the error message `e` provides almost no new information.

**Review by jkokosar — APPROVED** [review]

**Pull request merged**